Packages/Env/Sources/Env/StreamWatcher.swift

```diff
@@ -83,48 +83,42 @@ import OSLog
     }
   }
 
-  private func receiveMessage() {
-    task?.receive(completionHandler: { [weak self] result in
-      guard let self else { return }
-      switch result {
-      case let .success(message):
-        switch message {
-        case let .string(string):
-          do {
-            guard let data = string.data(using: .utf8) else {
-              logger.error("Error decoding streaming event string")
-              return
-            }
-            let rawEvent = try decoder.decode(RawStreamEvent.self, from: data)
-            logger.info("Stream update: \(rawEvent.event)")
-            if let event = rawEventToEvent(rawEvent: rawEvent) {
-              Task { @MainActor in
-                self.events.append(event)
-                self.latestEvent = event
-                if let event = event as? StreamEventNotification, event.notification.status?.visibility != .direct {
-                  self.unreadNotificationsCount += 1
-                }
-              }
+  private func receiveMessage() async {
+    do {
+      guard let message = try await task?.receive() else { return }
```

Comment:
I don't think you should return here as it could break the next receiveMessage loop.